### PR TITLE
python3Packages.bespon: 0.6.0 -> 0.7.0

### PR DESCRIPTION
**Pull request by chewblacka**
## Description of changes
https://github.com/gpoore/bespon_py/blob/master/CHANGELOG.md#v070-2023-10-15
Switched packaging to pyproject.toml.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by FliegendeWurst — APPROVED** [review]
Only change in this update is the switch to pyproject.toml

File: pkgs/development/python-modules/bespon/default.nix [hunk 1]
```diff
@@ -1,18 +1,20 @@
 { lib
 , buildPythonPackage
 , fetchPypi
+, python3Packages
```
Comment (NickCao) [inline pkgs/development/python-modules/bespon/default.nix:4]:
```suggestion
, setuptools
```

File: pkgs/development/python-modules/bespon/default.nix [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+  format = "pyproject";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "2f2bda67fea8ee95c8aa7e885835ab88bdbfa392a94077ce1c9d29017420ce7a";
+    hash = "sha256-dGtXw4uq6pdyXBVfSi9s7kCFUqA1PO7qWEGY0JNAz8Q=";
   };
 
-  propagatedBuildInputs = [ ];
+  propagatedBuildInputs = [ python3Packages.setuptools ];
```
Comment (NickCao) [inline pkgs/development/python-modules/bespon/default.nix:17]:
```suggestion
  nativeBuildInputs = [ setuptools ];
```

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 261822` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>codebraid</li>
    <li>codebraid.dist</li>
    <li>python310Packages.bespon</li>
    <li>python310Packages.bespon.dist</li>
    <li>python311Packages.bespon</li>
    <li>python311Packages.bespon.dist</li>
  </ul>
</details>

**Review by NickCao — APPROVED** [review]

**Pull request merged**